### PR TITLE
refactor appRevison extract func to utils

### PR DESCRIPTION
**Pull request by wangyikewxgm**
move ExtractRevisionNum func to utils/helper

### CODE REVIEW

File: pkg/oam/util/helper_test.go [hunk 1]
```diff
@@ -2054,3 +2054,29 @@ spec:
...
+	}, {
+		appRevision:     "v1-v10",
+		wantRevisionNum: "10",
+	}, {
+		appRevision:     "v10-v1-v1",
+		wantRevisionNum: "1",
+	}, {
+		appRevision:     "myapp-v1-v2",
+		wantRevisionNum: "2",
+	}}
```
Comment (wonderflow) [inline pkg/oam/util/helper_test.go:2077]: please add a case

```
		appRevision:     "v1",
		wantRevisionNum: "0",
```

File: pkg/oam/util/helper.go [hunk 2]
```diff
@@ -721,6 +721,13 @@ func ConvertComponentDef2WorkloadDef(dm discoverymapper.DiscoveryMapper, compone
 	return nil
 }
 
+// ExtractRevisionNum  extract revision number from appRevision name
+func ExtractRevisionNum(appRevision string) string {
```
Comment (wonderflow) [inline pkg/oam/util/helper.go:725]: the return value should be a int

File: pkg/oam/util/helper_test.go (hunk 1 above)
Comment (wonderflow) [reply pkg/oam/util/helper_test.go:2077]:
```
appRevision:     "vv",
wantRevisionNum: "0",
```

File: pkg/oam/util/helper.go (hunk 2 above)
Comment (wonderflow) [reply pkg/oam/util/helper.go:725]: we can return error if parse fail

Comment (wangyikewxgm) [reply pkg/oam/util/helper.go:725]: Got it

**Review by wonderflow — APPROVED** [review]
good job

**Comment by yangsoon** [comment]
LGTM

**Pull request merged**